src/models/Order.cpp

```diff
@@ -52,11 +52,12 @@ void Order::AcceptNewOrder(const Order& new_order) {
 
 void Order::UpdateOrder(const Order& order_update) {
   // Clear horizon.
-  order.edges.erase(remove_if(order.edges.begin(), order.edges.end(),
-      [](vda5050_msgs::Edge edge) { return !edge.released; }));
-  order.nodes.erase(remove_if(order.nodes.begin(), order.nodes.end(),
-      [](vda5050_msgs::Node node) { return !node.released; }));
-
+  auto it_edge = std::find_if(order.edges.begin(), order.edges.end(),
+      [](vda5050_msgs::Edge edge) { return !edge.released; });
+  order.edges.erase(it_edge, order.edges.end());
+  auto it_node = std::find_if(order.nodes.begin(), order.nodes.end(),
+      [](vda5050_msgs::Node node) { return !node.released; });
+  order.nodes.erase(it_node, order.nodes.end());
```

Comment:
Instead of find_if, use remove_if because it will sort the items in this list by putting the nodes/edges that are not released at the end of the list

Reply:
Find if will return the iterator pointing to the first unrelease node/edge, but here you don't have the guarantee if the next item in the list is released or not

Reply:
But according to the VDA5050, an item can't be released if the previous item is not released, am I wrong ?

Reply:
That's correct, in VDA, nodes and edges can not be released if the previous one is not released. But here I am referring to the array being used, and not the order graph.

For example, the node list can look like : 

```
"nodes": [
  {
    "nodeId": "7a434ec3-91c8-4334-89f8-31d30942fbb7",
    "sequenceId": 0,
    "nodeDescription": "Device initial position",
    "released": true,
    "nodePosition": {
      ...
    },
    "actions": [
      ...
    ]
  },
  {
    "nodeId": "11119fc4-7fe4-4a91-b6be-210a065131da",
    "sequenceId": 4,
    "nodeDescription": "",
    "released": false,
    "nodePosition": {
      ...
    },
    "actions": [
      ...
    ]
  },
  {
    "nodeId": "1c9f83e2-8478-46ae-8015-2cd321599aea",
    "sequenceId": 2,
    "nodeDescription": "",
    "released": true,
    "nodePosition": {
      ...
    },
    "actions": [
      ...
    ]
  }
]
```

Here you have 3 nodes, with nodes with sequenceId 0 and 2 being released, but 4 is not released. If you use find_if, the node with sequence id 4 will be found, and will cause the node with sequenceId 2 to be removed.

If you use remove_if, the node with sequenceId 4 will be put at the end (with other nodes with released = false) to be removed

Reply:
okay, I'll need to look into it again then. The problem is before I made those changes, the connector node kept crashed whenever an order update arrived. I've had the same issue when doing my own order update handling working with remove_if in the past.